src/test/java/me/moodcat/database/bootstrapper/BootstrapRule.java

```diff
@@ -0,0 +1,65 @@
+package me.moodcat.database.bootstrapper;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.moodcat.database.entities.Artist;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Method;
+
+/**
+ * A JUnit {@link MethodRule} that ensures a certain database environment is loaded
+ * for the test.
+ *
+ * @author Jan-Willem Gmelig Meyling
+ */
+@Singleton
+public class BootstrapRule implements MethodRule {
```

Comment:
I'm afraid not, because I read an annotation on the test method. The TestRules don't have access to the test methods to my knowledge. I know that the JUnit team is currently moving things around a bit for MethodRules. In our version of JUnit it is however not marked as deprecated, so I don't expect this to be removed soon.